packages/tree-select/src/index.js

```diff
@@ -79,7 +79,10 @@ const STATIC_FALSY_KEY = {};
  * The last map is a regular one because the the key for the last map is the string results of args.join().
  */
 function insertDependentKey( map, key, currentIndex, arr ) {
-	const weakMapKey = key || STATIC_FALSY_KEY;
+	if ( key != null && Object(key) !== key) {
```

Comment:
You'll need to disable the `eqeqeq` ESLint rule for the `!=` operator. And there should be extra whitespace inside `()` according to the WordPress style conventions. The wp-prettier pre-commit should have done that for you. 🤔

I think we're ready to merge after these nits are addressed. What do you think, is it ok to bump just the patch semver version for this change? It's a bugfix, but can break people's selectors if their state is messy.

Reply:
the linter isn't failing on it - i use `-n` to bypass the commit hooks since they don't work on my machine, and hooks are just for convenience, not enforcement.

I'd consider it a patch; all bugfixes of any kind are arguably breaking changes, so it's a judgement call :-)

Reply:
OK, `-n` explains everything, including the missing Prettier formatting that's also done by the pre-commit hook 🙂 Fixing the formatting is the only blocker at this moment.

> bypass the commit hooks since they don't work on my machine

Any details about that? Anything we can fix?

> hooks are just for convenience, not enforcement.

Ignoring the `eqeqeq` rule would mean that we ship one new ESLint warning, taking us further from the (very distant) goal of a zero-warnings build. Or would you recommend a different, more relaxed approach to linting?

Reply:
Hmm - I'm not suggesting ignoring any eslint errors, more saying that anything not enforced by CI - gating merges - is basically identical to having no checks in the first place. If `eqeqeq` is supposed to pass, then a failure should have failed this PR's CI checks.

So, first of all, I have husky globally disabled on my machine, since it's an incredibly dangerous tool - I've had it install git hooks in repos it shouldn't before, which can expose company secrets.

Looking at the hook error, though, it's because `prettier` isn't available; this is because `npm install` in the package directory doesn't install everything i need, and also because `npm install` at the root fails with a sass compilation error. I'm probably just using the wrong node version (v12 instead of v10, an easy thing to fix) but I also didn't try to fix it since the subproject I was working on (tree-select) didn't need sass. I've been running local tests with `npx jest packages/tree-select` and local linting with `npx eslint packages/tree-select`, which I would have expected to error on `eqeqeq` as well as prettier not being applied by `eslint-plugin-prettier` (ie, eslint handles prettier rather than formatting/linting being handled by N arbitrary tools).

I've switched to node 10, and `npm install` in the root works now; similarly, the commit hooks work now.

Clearly I need to check my assumptions about how projects are set up :-)

Reply:
> I have husky globally disabled on my machine, since it's an incredibly dangerous tool - I've had it install git hooks in repos it shouldn't before, which can expose company secrets.

Is it more dangerous than any other `npm install` script, which can also do almost anything?

> I'm probably just using the wrong node version (v12 instead of v10, an easy thing to fix)

Yes, we use the LTS version (i.e., 10.x) on the project. It would be nice to proactively figure out what exactly goes wrong with v12. As you mention SASS, and because `node-sass` installs and runs a compiled C binary, I suspect there are some binary incompatibilities when running `node-sass` with Node 12.